pomcorn/component.py

```diff
@@ -253,26 +270,27 @@ def all(self) -> list[ListItemType]:
         items: list[ListItemType] = []
         for locator in self.iter_locators(self.base_item_locator):
             items.append(
-                self._item_class(page=self.page, base_locator=locator),
+                self.item_class(page=self.page, base_locator=locator),
             )
         return items
 
+    @classmethod
+    def get_list_item_class(cls) -> type[ListItemType]:
+        """Return class passed in `Generic[ListItemType]`."""
+        return get_args(cls.__orig_bases__[0])[0]  # type: ignore
```

Comment:
Thanks to the example I see the problem. Could you try this?

```suggestion
    @classmethod
    def get_list_item_class(cls) -> type[ListItemType]:
        """Return class passed in `Generic[ListItemType]`."""
        base_class = next(
            klass
            for klass in cls.__orig_bases__  # type: ignore
            if inspect.isclass(typing.get_origin(klass))
            and issubclass(typing.get_origin(klass), ListComponent)
        )
        return get_args(base_class)[0]  # type: ignore
```

Reply:
Unfortunately I still get an wrong value on example code =(
![image](https://github.com/user-attachments/assets/0786dd71-1e78-4890-8047-18fa56ec2e7e)

`__orig_bases__` not store values of parent components
![image](https://github.com/user-attachments/assets/b0eba442-376f-4857-bad7-bfe459f71da7)

Reply:
It works, but not in a way I expected to) 
![image](https://github.com/user-attachments/assets/c5d88bc1-b025-4a22-a5a1-4f1eefcb5003)